### PR TITLE
[CA-1224] get google project name

### PR DESCRIPTION
**Pull request by andy7i**
https://broadworkbench.atlassian.net/browse/CA-1224

related to: https://github.com/broadinstitute/rawls/pull/1404
This adds code needed for an integration test.

**PR checklist**
- [x] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [x] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [x] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

### CODE REVIEW

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleProjectDAO.scala [hunk 1]
```diff
@@ -97,7 +97,7 @@ class HttpGoogleProjectDAO(appName: String,
       case e: HttpResponseException if e.getStatusCode == StatusCodes.NotFound.intValue => None
     } map {
       // return true if billing is enabled for the project, false otherwise
-      case Some(billingInfo) => billingInfo.getBillingEnabled
+      case Some(billingInfo) => billingInfo.getBillingEnabled == true
```
Comment (andy7i) [inline google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleProjectDAO.scala:101]: `billingInfo.getBillingEnabled` returns null or boolean, so this should eliminate the possibility of a NPE

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleProjectDAO.scala [hunk 2]
```diff
@@ -134,4 +134,12 @@ class HttpGoogleProjectDAO(appName: String,
...
       case e: HttpResponseException if e.getStatusCode == StatusCodes.NotFound.intValue => None
     }
+
+  override def getProjectName(projectId: String): Future[Option[String]] =
+    retryWithRecover(when5xx, whenUsageLimited, whenInvalidValueOnBucketCreation, whenNonHttpIOException) { () =>
+      Option(executeGoogleRequest(cloudResManager.projects().get(projectId))).map(_.getName)
+    } {
+      // if the project doesn't exist, don't fail
+      case e: HttpResponseException if e.getStatusCode == StatusCodes.NotFound.intValue => None
+    }
```
Comment (andy7i) [inline google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleProjectDAO.scala:145]: this follows the same structure as other functions in this class.

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleProjectDAO.scala (hunk 1 above)
Comment (gpolumbo-broad) [reply google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleProjectDAO.scala:101]: Recommend putting this comment in the code

**Review by gpolumbo-broad — APPROVED** [review]

**Review by s-rubenstein — APPROVED** [review]
LGTM

**Pull request merged**